### PR TITLE
Add typeclass instances for Map (Issue #54)

### PR DESCRIPTION
**Pull request by bobbyrauchenberg**
Adds the following typeclasses to map
* Traverse
* Flatmap

### CODE REVIEW

**Comment by bobbyrauchenberg** [comment]
I should probably cleanup the type lambdas in here, if someone can give me a pointer on the preferred way to do this

File: std/src/main/scala/cats/std/map.scala [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package cats
+package std
+
+trait MapInstances {
+
+  implicit def mapInstance[K]: Traverse[({type F[V] = Map[K,V]})#F] with FlatMap[({type F[V] = Map[K,V]})#F] =
+    new Traverse[({type F[V] = Map[K,V]})#F] with FlatMap[({type F[V] = Map[K,V]})#F] {
```
Comment (non) [inline std/src/main/scala/cats/std/map.scala]: You can write `({type F[V] = Map[K,V]})#F` as `Map[K, ?]` here.

**Comment by julien-truffaut** [comment]
seems to be a duplicate of #65 

I think we should write a note in the original ticket in order to avoid that, e.g. "Working on it" 

@bobbyrauchenberg you can use kind projector to replace type lamda https://github.com/non/kind-projector

**Comment by bobbyrauchenberg** [comment]
@julien-truffaut that's pretty cool. thanks, i've updated the PR.

**Comment by ceedubs** [comment]
@bobbyrauchenberg thanks a lot for putting this together! Since this pull request is almost identical to #65 but that one already has a `Show` instance already on it, I think it might make sense to close this out as a duplicate of that one. Is that okay? I'm sorry - we didn't expect contributions to start coming in so fast and we hadn't yet set up a process to keep people from duplicating effort :(. Hopefully this will be less of a problem now that we address it in the README and contributor guide.

**Comment by bobbyrauchenberg** [comment]
@ceedubs of course, that's no problem. I should have checked before I started

**Comment by ceedubs** [comment]
@bobbyrauchenberg thanks for being so understanding. And the two PRs were back-to-back, so the other probably hadn't even been submitted when you started your work!

**Pull request closed without merging**